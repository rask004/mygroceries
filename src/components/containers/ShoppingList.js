import React, { Component } from 'react';
import * as D from 'date-fns';
import { connect } from 'react-redux';
import {withRouter} from 'react-router-dom';
import PropTypes from 'prop-types';
import { addShoppingDay, removeShoppingDay, 
    removeShoppingItem, addShoppingItem } from '../../store/actions';
import Card from '@material-ui/core/Card';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import Grid from '@material-ui/core/Grid';
import {FaTrash, FaPlus} from 'react-icons/fa';
import ShoppingListDialog, {ShoppingListPanel} from './NewShoppingDialog';
import 'typeface-roboto';
import '../../css/shoppinglist.css';


class ShoppingList extends Component {

    state = {
        showModal: false
    }

    toggleShowModal = () => {
        this.setState((prevState) => {return {showModal: !prevState.showModal}});
    }

    render() {
        const products = this.props.products;
        const shoppinglists = this.props.shoppingLists.length > 0 ?
        this.props.shoppingLists.map( (shoplist, step) => {
            const items = shoplist.items.slice();
            const shoppingitems = items.length > 0 ?
            items.map( (item) => {
                const product = products.find(p => p.id === item.id);
                return (
                    <li key={item.id}>
                        <Grid container>
                            <Grid item xs={12} sm={6} md={3}>
                                <Typography component="span" className="content-brand">
                                    {product.brand}
                                </Typography>
                            </Grid>
                            <Grid item xs={12} sm={6} md={4}>
                                <Typography component="span" className="content-name">
                                    {product.name}
                                </Typography>
                            </Grid>
                            <Grid item xs={12} sm={6} md={1}>
                                <Typography component="span" className="content-quantity" align="right">
                                    {item.quantity}
                                </Typography>
                            </Grid>
                            <Grid item xs={12} sm={6} md={1}>
                                <Typography component="span" className="content-unit">
                                    {item.unit}
                                </Typography>
                            </Grid>
                            <Grid item xs={12} sm={6} md={2}>
                                <Typography component="span" className="content-price" align="right">
                                    -unit * quantity price-
                                </Typography>
                            </Grid>
                            <Grid item xs={12} sm={12} md={1} align="right">
                                <Button variant="contained" size="small" color="primary" type="button"
                                    className="remove-item-action"
                                    onClick={() => this.props.onRemoveShoppingListItem(shoplist.datetime, item.id)}>
                                    <FaTrash/>
                                </Button>
                            </Grid>
                        </Grid>
                    </li>
                );
            }) :
            <Grid container>
                <Grid item xs={12}>
                    <Typography component="span">This list has no items.</Typography>
                </Grid>
            </Grid>;

            const summaryItem = items.length > 0 ?
            <li key={-1}>
                <Grid container>
                    <Grid item xs={12} sm={6} md={9} />
                    <Grid item xs={12} sm={6} md={2} align="right">
                        <Typography className="content-total" align="right">
                            -total price-
                        </Typography>
                    </Grid> 
                </Grid>
            </li> :
            "";

            return (
                <div key={shoplist.datetime} className="content-shopping">
                    <div className="content-title">
                        <Grid container>
                            <Grid item xs={12} sm={6} md={11}>
                                <Typography className="content-date" variant="title">
                                    {D.format(shoplist.datetime, "dddd, Do MMMM")}
                                </Typography>
                            </Grid>
                            <Grid item xs={12} sm={6} md={1} align="right">
                                <Button variant="contained" size="medium" color="primary" type="button"
                                    className="remove-list-action"
                                    onClick={() => this.props.onRemoveShoppingList(shoplist.datetime)}>
                                    <FaTrash/>
                                </Button>
                            </Grid>
                        </Grid>
                    </div>
                    
                    <ul className="content-items">
                        {shoppingitems}
                        {summaryItem}
                    </ul>
                </div>
            )
        }) :
        <div className="content-notice">
            <Typography component="span">
                No shopping lists are present at this time.
            </Typography>
        </div>;

        return (
            <React.Fragment>
                <Card className="shopping-list">
                    <Typography className="shopping-title" variant="display1" 
                        gutterBottom={true}>
                        Shopping Lists
                    </Typography>
                    {shoppinglists}
                    <div className="content-add-section">
                        <Button className="action-add" variant="contained"
                            size="large" color="primary" type="button" 
                            onClick={this.toggleShowModal}>
                            Add...
                        </Button>
                    </div>
                </Card>
                <ShoppingListDialog
                    show={this.state.showModal}
                    onClose={this.toggleShowModal}
                    products={this.props.products}
                    shoppingLists={this.props.shoppingLists}
                    addList={this.props.onAddShoppingList}
                    addListItem={this.props.onAddShoppingListItem}
                />
            </React.Fragment>
        );
    }
}
ShoppingList.propTypes = {
    mealplans: PropTypes.array.isRequired,
    recipes: PropTypes.array.isRequired,
    products: PropTypes.array.isRequired,
    shoppingLists: PropTypes.array.isRequired,
    onAddShoppingList: PropTypes.func.isRequired,
    onRemoveShoppingList: PropTypes.func.isRequired,
    onAddShoppingListItem: PropTypes.func.isRequired,
    onRemoveShoppingListItem: PropTypes.func.isRequired,
}


const mapStateToProps = state => {
	return {
        mealplans: state.mealplans,
        recipes: state.recipes,
        products: state.products,
        shoppingLists: state.shoppinglists
    }
}

const mapDispatchToProps = dispatch => {
	return {
		onAddShoppingList(datetime, recurring, frequencyType, frequencyRate) {
			dispatch(
				addShoppingDay(datetime, recurring, frequencyType, frequencyRate)
			);
        },
        onRemoveShoppingList(datetime) {
            dispatch(
                removeShoppingDay(datetime)
            );
        },
        onAddShoppingListItem(datetime, item) {
            dispatch(
                addShoppingItem(datetime, item)
            );
        },
        onRemoveShoppingListItem(datetime, id) {
            dispatch(
                removeShoppingItem(datetime, id)
            );
        }
	}
}

export {ShoppingList};

const ConnectedShoppingList = connect(mapStateToProps, mapDispatchToProps)(ShoppingList);
export default withRouter(ConnectedShoppingList);