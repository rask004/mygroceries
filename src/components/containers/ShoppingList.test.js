import React from 'react';
import {create} from 'react-test-renderer';
import * as D from 'date-fns';

import {ShoppingList} from './ShoppingList';

import sampleData from '../../data/initialState.json';


const mealplans = sampleData.mealplans;
const recipes = sampleData.recipes;
const products = sampleData.products;
const shoppinglists = sampleData.shoppinglists;

const shoppingList = <ShoppingList
        mealplans={mealplans}
        recipes={recipes}
        products={products}
        shoppingLists={shoppinglists}
        onAddShoppingList={jest.fn()}
        onRemoveShoppingList={jest.fn()}
        onAddShoppingListItem={jest.fn()}
        onRemoveShoppingListItem={jest.fn()}
    />  


describe('ShoppingList Component rendering tests', () => {
    test('rendering snapshot', () => {
        const component = create(shoppingList);
        expect(component.toJSON()).toMatchSnapshot();
    });

    test('individual Shopping Lists are rendered correctly', () => {
        const expectedListItems = sampleData.shoppinglists.slice();
        const products = sampleData.products.slice();

        const component = create(shoppingList).root;

        const shoppinglists = component.findAllByProps({className:"content-shopping"});
        shoppinglists.forEach( (shoplist, index) => {
            const expectedDate = D.format(expectedListItems[index].datetime, "dddd, Do MMMM");
            const shownDate = shoplist
                .findByProps({className:"content-date"}).props.children;
            expect(shownDate)
                .toEqual(expectedDate);
            const expectedItems = expectedListItems[index].items;
            const shoplistItems = shoplist.findAllByType("li");
            const items = shoplistItems.slice(shoplistItems.length);

            // last "item" is actually a summary of costs.
            items.forEach( (shoplistItem, index) => {
                const expecteditem = expectedItems[index];
                const expectedItemId = expecteditem.id;
                const expectedProduct = products.find(item => item.id === expectedItemId);
                const quantity = shoplistItem.findByProps({className:"content-quantity"}).props.children;
                const unit = shoplistItem.findByProps({className:"content-unit"}).props.children;
                const name = shoplistItem.findByProps({className:"content-name"}).props.children;
                const brand = shoplistItem.findByProps({className:"content-brand"}).props.children;
                expect(Number(quantity)).toBe(expecteditem.quantity);
                expect(unit).toBe(expecteditem.unit);
                expect(name).toBe(expectedProduct.name);
                expect(brand).toBe(expectedProduct.brand);
            });
        });
    });
});

describe('ShoppingList Component behaviour tests', () => {

    test('click add ShoppingList calls onAddShoppingList', () => {
        expect(false).toBeTruthy();
    });

    test('click remove ShoppingList calls onRemoveShoppingList', () => {
        const expectedList = sampleData.shoppinglists.slice()[0];
        const component = create(shoppingList).root;
        const shoppinglist = component.findAllByProps({className:"content-shopping"})[0];
        const removeListButton = shoppinglist.findByProps({className: "remove-list-action"});
        removeListButton.props.onClick();
        expect(component.props.onRemoveShoppingList).toHaveBeenCalledWith(expectedList.datetime);
    });

    test('click add ShoppingList item calls onAddShoppingListItem', () => {
        expect(false).toBeTruthy();
    });

    test('click remove ShoppingList item calls onRemoveShoppingListItem', () => {
        const component = create(shoppingList).root;
        const expectedList = sampleData.shoppinglists.slice()[0];
        const shoppinglist = component.findAllByProps({className:"content-shopping"})[0];
        const expectedItemRemoved = expectedList.items[0];
        const shoplistItem = shoppinglist.findAllByType("li")[0];
        const removeItemButton = shoplistItem.findByProps({className: "remove-item-action"});
        removeItemButton.props.onClick();
        expect(component.props.onRemoveShoppingListItem).toHaveBeenCalledWith(expectedList.datetime, expectedItemRemoved.id);
    });
});
