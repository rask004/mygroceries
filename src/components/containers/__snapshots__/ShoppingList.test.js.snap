// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShoppingList Component rendering tests rendering snapshot 1`] = `
<div
  className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1 shopping-list"
>
  <h1
    className="MuiTypography-root-29 MuiTypography-display1-33 MuiTypography-gutterBottom-56 shopping-title"
  >
    Shopping Lists
  </h1>
  <div
    className="content-shopping"
  >
    <div
      className="MuiGrid-container-65"
    >
      <div
        className="MuiGrid-item-66 MuiGrid-grid-xs-12-105 MuiGrid-grid-sm-6-113 MuiGrid-grid-md-11-132"
      >
        <h2
          className="MuiTypography-root-29 MuiTypography-title-35 content-date"
        >
          Saturday, 14th July
        </h2>
      </div>
      <div
        align="right"
        className="MuiGrid-item-66 MuiGrid-grid-xs-12-105 MuiGrid-grid-sm-6-113 MuiGrid-grid-md-1-122"
      >
        <button
          className="MuiButtonBase-root-188 MuiButton-root-162 MuiButton-contained-173 MuiButton-containedPrimary-174 MuiButton-raised-176 MuiButton-raisedPrimary-177 remove-list-action"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiButton-label-163"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-191"
          />
        </button>
      </div>
    </div>
    <ul
      className="content-items"
    >
      <li>
        <div
          className="MuiGrid-container-65"
        >
          <div
            className="MuiGrid-item-66 MuiGrid-grid-xs-12-105 MuiGrid-grid-sm-6-113 MuiGrid-grid-md-3-124"
          >
            <span
              className="MuiTypography-root-29 MuiTypography-body1-38 content-name"
            >
              orange
            </span>
          </div>
          <div
            className="MuiGrid-item-66 MuiGrid-grid-xs-12-105 MuiGrid-grid-sm-6-113 MuiGrid-grid-md-2-123"
          >
            <span
              className="MuiTypography-root-29 MuiTypography-body1-38 content-brand"
            >
              Pure Orchards LTD
            </span>
          </div>
          <div
            className="MuiGrid-item-66 MuiGrid-grid-xs-12-105 MuiGrid-grid-sm-6-113 MuiGrid-grid-md-2-123"
          >
            <span
              className="MuiTypography-root-29 MuiTypography-body1-38 content-country"
            >
              New Zealand
            </span>
          </div>
          <div
            className="MuiGrid-item-66 MuiGrid-grid-xs-12-105 MuiGrid-grid-sm-6-113 MuiGrid-grid-md-1-122"
          >
            <span
              className="MuiTypography-root-29 MuiTypography-body1-38 MuiTypography-alignRight-53 content-quantity"
            >
              5
            </span>
          </div>
          <div
            className="MuiGrid-item-66 MuiGrid-grid-xs-12-105 MuiGrid-grid-sm-6-113 MuiGrid-grid-md-1-122"
          >
            <span
              className="MuiTypography-root-29 MuiTypography-body1-38 content-unit"
            >
              each
            </span>
          </div>
          <div
            className="MuiGrid-item-66 MuiGrid-grid-xs-12-105 MuiGrid-grid-sm-6-113 MuiGrid-grid-md-2-123"
          >
            <span
              className="MuiTypography-root-29 MuiTypography-body1-38 MuiTypography-alignRight-53 content-price"
            >
              -unit price-
            </span>
          </div>
          <div
            align="right"
            className="MuiGrid-item-66 MuiGrid-grid-xs-12-105 MuiGrid-grid-sm-12-119 MuiGrid-grid-md-1-122"
          >
            <button
              className="MuiButtonBase-root-188 MuiButton-root-162 MuiButton-contained-173 MuiButton-containedPrimary-174 MuiButton-raised-176 MuiButton-raisedPrimary-177 MuiButton-sizeSmall-185 remove-item-action"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="0"
              type="button"
            >
              <span
                className="MuiButton-label-163"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 448 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
                  />
                </svg>
              </span>
              <span
                className="MuiTouchRipple-root-191"
              />
            </button>
          </div>
        </div>
      </li>
      <li>
        <div
          className="MuiGrid-container-65"
        >
          <div
            className="MuiGrid-item-66 MuiGrid-grid-xs-12-105 MuiGrid-grid-sm-6-113 MuiGrid-grid-md-3-124"
          >
            <span
              className="MuiTypography-root-29 MuiTypography-body1-38 content-name"
            >
              apple
            </span>
          </div>
          <div
            className="MuiGrid-item-66 MuiGrid-grid-xs-12-105 MuiGrid-grid-sm-6-113 MuiGrid-grid-md-2-123"
          >
            <span
              className="MuiTypography-root-29 MuiTypography-body1-38 content-brand"
            >
              San Fernando Produce
            </span>
          </div>
          <div
            className="MuiGrid-item-66 MuiGrid-grid-xs-12-105 MuiGrid-grid-sm-6-113 MuiGrid-grid-md-2-123"
          >
            <span
              className="MuiTypography-root-29 MuiTypography-body1-38 content-country"
            >
              California
            </span>
          </div>
          <div
            className="MuiGrid-item-66 MuiGrid-grid-xs-12-105 MuiGrid-grid-sm-6-113 MuiGrid-grid-md-1-122"
          >
            <span
              className="MuiTypography-root-29 MuiTypography-body1-38 MuiTypography-alignRight-53 content-quantity"
            >
              3
            </span>
          </div>
          <div
            className="MuiGrid-item-66 MuiGrid-grid-xs-12-105 MuiGrid-grid-sm-6-113 MuiGrid-grid-md-1-122"
          >
            <span
              className="MuiTypography-root-29 MuiTypography-body1-38 content-unit"
            >
              each
            </span>
          </div>
          <div
            className="MuiGrid-item-66 MuiGrid-grid-xs-12-105 MuiGrid-grid-sm-6-113 MuiGrid-grid-md-2-123"
          >
            <span
              className="MuiTypography-root-29 MuiTypography-body1-38 MuiTypography-alignRight-53 content-price"
            >
              -unit price-
            </span>
          </div>
          <div
            align="right"
            className="MuiGrid-item-66 MuiGrid-grid-xs-12-105 MuiGrid-grid-sm-12-119 MuiGrid-grid-md-1-122"
          >
            <button
              className="MuiButtonBase-root-188 MuiButton-root-162 MuiButton-contained-173 MuiButton-containedPrimary-174 MuiButton-raised-176 MuiButton-raisedPrimary-177 MuiButton-sizeSmall-185 remove-item-action"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="0"
              type="button"
            >
              <span
                className="MuiButton-label-163"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 448 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
                  />
                </svg>
              </span>
              <span
                className="MuiTouchRipple-root-191"
              />
            </button>
          </div>
        </div>
      </li>
      <li>
        <div
          className="MuiGrid-container-65"
        >
          <div
            className="MuiGrid-item-66 MuiGrid-grid-xs-12-105 MuiGrid-grid-sm-6-113 MuiGrid-grid-md-9-130"
          />
          <div
            align="right"
            className="MuiGrid-item-66 MuiGrid-grid-xs-12-105 MuiGrid-grid-sm-6-113 MuiGrid-grid-md-2-123"
          >
            <p
              className="MuiTypography-root-29 MuiTypography-body1-38 MuiTypography-alignRight-53 content-total"
            >
              -total price-
            </p>
          </div>
        </div>
      </li>
    </ul>
  </div>
  <div
    className="content-shopping"
  >
    <div
      className="MuiGrid-container-65"
    >
      <div
        className="MuiGrid-item-66 MuiGrid-grid-xs-12-105 MuiGrid-grid-sm-6-113 MuiGrid-grid-md-11-132"
      >
        <h2
          className="MuiTypography-root-29 MuiTypography-title-35 content-date"
        >
          Tuesday, 17th July
        </h2>
      </div>
      <div
        align="right"
        className="MuiGrid-item-66 MuiGrid-grid-xs-12-105 MuiGrid-grid-sm-6-113 MuiGrid-grid-md-1-122"
      >
        <button
          className="MuiButtonBase-root-188 MuiButton-root-162 MuiButton-contained-173 MuiButton-containedPrimary-174 MuiButton-raised-176 MuiButton-raisedPrimary-177 remove-list-action"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiButton-label-163"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-191"
          />
        </button>
      </div>
    </div>
    <ul
      className="content-items"
    >
      <li>
        <div
          className="MuiGrid-container-65"
        >
          <div
            className="MuiGrid-item-66 MuiGrid-grid-xs-12-105 MuiGrid-grid-sm-6-113 MuiGrid-grid-md-3-124"
          >
            <span
              className="MuiTypography-root-29 MuiTypography-body1-38 content-name"
            >
              anaroba extra-strong coffee
            </span>
          </div>
          <div
            className="MuiGrid-item-66 MuiGrid-grid-xs-12-105 MuiGrid-grid-sm-6-113 MuiGrid-grid-md-2-123"
          >
            <span
              className="MuiTypography-root-29 MuiTypography-body1-38 content-brand"
            >
              Rio de Feliz Comidas
            </span>
          </div>
          <div
            className="MuiGrid-item-66 MuiGrid-grid-xs-12-105 MuiGrid-grid-sm-6-113 MuiGrid-grid-md-2-123"
          >
            <span
              className="MuiTypography-root-29 MuiTypography-body1-38 content-country"
            >
              Argentina
            </span>
          </div>
          <div
            className="MuiGrid-item-66 MuiGrid-grid-xs-12-105 MuiGrid-grid-sm-6-113 MuiGrid-grid-md-1-122"
          >
            <span
              className="MuiTypography-root-29 MuiTypography-body1-38 MuiTypography-alignRight-53 content-quantity"
            >
              300
            </span>
          </div>
          <div
            className="MuiGrid-item-66 MuiGrid-grid-xs-12-105 MuiGrid-grid-sm-6-113 MuiGrid-grid-md-1-122"
          >
            <span
              className="MuiTypography-root-29 MuiTypography-body1-38 content-unit"
            >
              grams
            </span>
          </div>
          <div
            className="MuiGrid-item-66 MuiGrid-grid-xs-12-105 MuiGrid-grid-sm-6-113 MuiGrid-grid-md-2-123"
          >
            <span
              className="MuiTypography-root-29 MuiTypography-body1-38 MuiTypography-alignRight-53 content-price"
            >
              -unit price-
            </span>
          </div>
          <div
            align="right"
            className="MuiGrid-item-66 MuiGrid-grid-xs-12-105 MuiGrid-grid-sm-12-119 MuiGrid-grid-md-1-122"
          >
            <button
              className="MuiButtonBase-root-188 MuiButton-root-162 MuiButton-contained-173 MuiButton-containedPrimary-174 MuiButton-raised-176 MuiButton-raisedPrimary-177 MuiButton-sizeSmall-185 remove-item-action"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="0"
              type="button"
            >
              <span
                className="MuiButton-label-163"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 448 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
                  />
                </svg>
              </span>
              <span
                className="MuiTouchRipple-root-191"
              />
            </button>
          </div>
        </div>
      </li>
      <li>
        <div
          className="MuiGrid-container-65"
        >
          <div
            className="MuiGrid-item-66 MuiGrid-grid-xs-12-105 MuiGrid-grid-sm-6-113 MuiGrid-grid-md-9-130"
          />
          <div
            align="right"
            className="MuiGrid-item-66 MuiGrid-grid-xs-12-105 MuiGrid-grid-sm-6-113 MuiGrid-grid-md-2-123"
          >
            <p
              className="MuiTypography-root-29 MuiTypography-body1-38 MuiTypography-alignRight-53 content-total"
            >
              -total price-
            </p>
          </div>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
